skrub/_datetime_encoder.py

```diff
@@ -18,22 +16,56 @@
     "microsecond": "us",
     "nanosecond": "N",
 }
-TIME_LEVELS: list[str] = list(WORD_TO_ALIAS.keys())
-AcceptedTimeValues = Literal[
-    "year",
-    "month",
-    "day",
-    "hour",
-    "minute",
-    "second",
-    "microsecond",
-    "nanosecond",
-]
-
-
-class DatetimeEncoder(BaseEstimator, TransformerMixin):
-    """Transform each datetime column into several numeric columns \
-    for temporal features (e.g. "year", "month", "day"...).
+TIME_LEVELS = list(WORD_TO_ALIAS)
+
+
+def is_datetime_parsable(X):
+    """Check whether a 1d vector can be converted into a \
+    :class:`~pandas.core.indexes.datetimes.DatetimeIndex`.
+
+    Parameters
+    ----------
+    X : array-like of shape ``(n_sample,)``
+
+    Returns
+    -------
+    is_dt_parsable : bool
+    """
+    if len(X.shape) > 1:
+        raise ValueError(f"X must be 1d, got shape: {X.shape}.")
+    np_dtypes_candidates = [np.object_, np.str_, np.datetime64]
+    if any(np.issubdtype(X.dtype, np_dtype) for np_dtype in np_dtypes_candidates):
+        try:
+            _ = pd.to_datetime(X)
```

Comment:
I think `to_datetime` is stricter than `DateTimeIndex` which was used before (@LeoGrin  also mentioned that in #768 [here](https://github.com/skrub-data/skrub/issues/746#issuecomment-1731417491)), as a result the docstring [example](https://github.com/skrub-data/skrub/pull/784/files#diff-af731509aaef29b0ef5b34bb9bd06c8956a0f1112e0b40218c5ea7c460b4c6a9L94) prints an empty vectorized dataframe on my machine. 

being a bit more strict is probably a good idea especially at first; using `to_datetime` instead of `DateTimeIndex` avoids issues such as [this one](https://github.com/skrub-data/skrub/issues/768#issuecomment-1742753260), but the example needs to be updated